### PR TITLE
[Activation] Remove block mined from context

### PR DESCRIPTION
**Pull request by noescape00**
This PR removes .BlockMined property from the context and fixes mempool tests + mocking that fails CHT tests + improves stability of BS tests

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Consensus.Tests/TestChainFactory.cs [hunk 1]
```diff
@@ -144,7 +144,7 @@ public static async Task<TestChainContext> CreateAsync(Network network, string d
             testChainContext.Consensus = new ConsensusManager(network, testChainContext.LoggerFactory, testChainContext.ChainState, testChainContext.HeaderValidator, testChainContext.IntegrityValidator,
                 testChainContext.PartialValidator, testChainContext.FullValidator, testChainContext.Checkpoints, consensusSettings, testChainContext.ConsensusRules, testChainContext.FinalizedBlockInfo, new Signals.Signals(),
                 testChainContext.PeerBanning, testChainContext.InitialBlockDownloadState, testChainContext.Chain, new Mock<IBlockPuller>().Object, blockStore,
-                new InvalidBlockHashStore(new DateTimeProvider()), new Mock<ConnectionManager>().Object);
+                new InvalidBlockHashStore(new DateTimeProvider()), new Mock<IConnectionManager>().Object);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Consensus.Tests/TestChainFactory.cs]: Ah nice, I also saw this bug in the tests ;)

File: src/Stratis.Bitcoin.Features.MemoryPool.Tests/TestChainFactory.cs [hunk 2]
```diff
@@ -77,16 +78,20 @@ public static async Task<ITestChainContext> CreateAsync(Network network, Script
             network.Consensus.Options = new ConsensusOptions();
             new FullNodeBuilderConsensusExtension.PowConsensusRulesRegistration().RegisterRules(network.Consensus);
 
+            // Dont check PoW of a header in this test.
+            IHeaderValidationConsensusRule checkDiffRule = network.Consensus.HeaderValidationRules.SingleOrDefault(x => x.GetType() == typeof(CheckDifficultyPowRule));
```
Comment (ferdeen) [inline src/Stratis.Bitcoin.Features.MemoryPool.Tests/TestChainFactory.cs]: Could do `network.Consensus.HeaderValidationRules.RemoveAll(x => x.GetType() == typeof(CheckDifficultyPowRule));` here.

File: src/Stratis.Bitcoin.Features.MemoryPool.Tests/TestChainFactory.cs [hunk 3]
```diff
@@ -77,16 +78,20 @@ public static async Task<ITestChainContext> CreateAsync(Network network, Script
...
+            network.Consensus.HeaderValidationRules.Remove(checkDiffRule);
```
Comment (ferdeen) [inline src/Stratis.Bitcoin.Features.MemoryPool.Tests/TestChainFactory.cs]: Could do network.Consensus.HeaderValidationRules.RemoveAll(x => x.GetType() == typeof(CheckDifficultyPowRule)); here.

Comment (noescape00) [reply src/Stratis.Bitcoin.Features.MemoryPool.Tests/TestChainFactory.cs]: fixed

File: src/Stratis.Bitcoin.Features.MemoryPool.Tests/TestChainFactory.cs [hunk 4]
```diff
@@ -77,16 +78,19 @@ public static async Task<ITestChainContext> CreateAsync(Network network, Script
             network.Consensus.Options = new ConsensusOptions();
             new FullNodeBuilderConsensusExtension.PowConsensusRulesRegistration().RegisterRules(network.Consensus);
 
+            // Dont check PoW of a header in this test.
+            network.Consensus.HeaderValidationRules.RemoveAll(x => x.GetType() == typeof(CheckDifficultyPowRule));
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.MemoryPool.Tests/TestChainFactory.cs]: Interesting approach indeed, I like the creative thinking and the correct usage of rules

File: src/Stratis.Bitcoin.Features.MemoryPool.Tests/TestChainFactory.cs (hunk 3 above)
Comment (noescape00) [reply src/Stratis.Bitcoin.Features.MemoryPool.Tests/TestChainFactory.cs]: yes. but that will also require 2 lines. one liner is shorter.

**Pull request merged**